simulation-backend/simulation_backend/DEinterface.py

```diff
@@ -1,1856 +1,145 @@
-                ].append(
-                    {
-                        "data": edc_detail.tolist(),
-                        # "data_pressure": pressure_detail.tolist(),
-                        "t": (t_off - t_off[0]).tolist(),
-                        "frequency": result_container["results"][0]["frequencies"][
-                            index
-                        ],
-                        "type": "edc",
-                    }
-                )
-                if "t" not in df.columns:
-                    df["t"] = (t_off - t_off[0]).tolist()
-                df[str(result_container["results"][0]["frequencies"][index]) + "Hz"] = (
-                    pressure_detail.tolist()
-                )
-
-        with open(json_file_path, "w") as new_result_json:
-            new_result_json.write(json.dumps(result_container, indent=4))
-        with open(
-            json_file_path.replace(".json", "_pressure.csv"), "w", newline=""
-        ) as pressure_result_csv:
-            df.to_csv(pressure_result_csv, index=False)
-
-    et = time.time()  # end time
-    elapsed_time = et - st
-
-    # %%
-    ###############################################################################
-    # SAVING
-    ###############################################################################
-    # np.save(os.path.join('dt'),dt)
-    # np.save(os.path.join('w_rec_off_band'),w_rec_off_band)
-    # np.save(os.path.join('w_rec_off_deriv_band'),w_rec_off_deriv_band)
-    # np.save(os.path.join('p_rec_off_deriv_band'),p_rec_off_deriv_band)
-    # np.save(os.path.join('t_off'),t_off)
-
-    # Save all variables to a file
-    def save(filename):
-        with open(filename, "wb") as f:
-            # Filter out modules, functions, and other unsupported types
-            filtered_variables = {}
-            for k, v in globals().items():
-                try:
-                    # Check if the object can be pickled
-                    pickle.dumps(v)
-                    # Exclude some types explicitly known to cause issues
-                    if not k.startswith("__") and not isinstance(
-                        v,
-                        (
-                            types.ModuleType,
-                            types.FunctionType,
-                            types.BuiltinFunctionType,
-                            types.LambdaType,
-                            types.MethodType,
-                            types.MappingProxyType,
-                        ),
-                    ):
-                        filtered_variables[k] = v
-                except Exception as e:
-                    print(f"Could not pickle {k}: {str(e)}")
-
-            pickle.dump(filtered_variables, f)
-
-    # To save all current variables
-    save("results.pkl")
-    gmsh.finalize()
+    except Exception as e:
+        print("Error saving the simulation solver settings")
+        raise Exception(f"Error saving the simulation solver settings: {e}")
```

Comment:
```suggestion
        raise Exception("Error saving the simulation solver settings:") from e
```
Here it would make sense to raise a more specific type of exception and re-raise it from e